freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/ums/EnvironmentsRightsChecksFactory.java

```diff
@@ -0,0 +1,20 @@
+package com.sequenceiq.freeipa.service.freeipa.user.ums;
+
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto;
+import com.sequenceiq.freeipa.service.freeipa.user.UserSyncConstants;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class EnvironmentsRightsChecksFactory {
```

Comment:
so there is a `EnvironmentRightChecksFactory`, and I'm a bit confused at first blink which is which and why.

Reply:
The two providers use different mechanisms to determine rights. 
The default provider uses the RightChecks model to call the authorization service.
the bulk provider uses the RightsChecks model to call the user management service.
It's unfortunate that the two are not the same, and even more unfortunate that they are so similarly named.
I can rename these classes to better indicate the auth vs ums difference.

Reply:
renamed to `AuthorizationRightChecksFactory` and `UmsRightsChecksFactory`